pkg/karmadactl/logs.go

```diff
@@ -50,18 +50,20 @@ var (
 )
 
 // NewCmdLogs new logs command.
-func NewCmdLogs(karmadaConfig KarmadaConfig, parentCommand string, streams genericclioptions.IOStreams) *cobra.Command {
+// TODO(@carlory): take a factory as an argument when we resolve conflicts on flags.
```

Comment:
Could we directly resolve this `TODO` in this PR? Move `f := util.NewFactory(defaultConfigFlags)` to `pkg/karmadactl`?

Reply:
@lonelyCZ This requires changing all subcommands,  I filled #2349 to track this task. 

```
(⎈ |karmada:default)➜  karmada git:(karmadactl-factory) ✗ go run cmd/karmadactl/karmadactl.go
panic: karmadactl flag redefined: kubeconfig

goroutine 1 [running]:
github.com/spf13/pflag.(*FlagSet).AddFlag(0x140000de600, 0x14000924b40)
	/Users/kiki/workspace/golang/src/github.com/karmada-io/karmada/vendor/github.com/spf13/pflag/flag.go:848 +0x548
github.com/spf13/pflag.(*FlagSet).VarPF(0xa?, {0x106849978, 0x140003df460}, {0x106049248, 0xa}, {0x0, 0x0}, {0x10609b245, 0x34})
	/Users/kiki/workspace/golang/src/github.com/karmada-io/karmada/vendor/github.com/spf13/pflag/flag.go:831 +0x140
github.com/spf13/pflag.(*FlagSet).VarP(...)
	/Users/kiki/workspace/golang/src/github.com/karmada-io/karmada/vendor/github.com/spf13/pflag/flag.go:837
github.com/spf13/pflag.(*FlagSet).StringVar(0x3?, 0x1020100?, {0x106049248?, 0xffffffffffffffff?}, {0x0?, 0x10602b450?}, {0x10609b245?, 0x10602b448?})
	/Users/kiki/workspace/golang/src/github.com/karmada-io/karmada/vendor/github.com/spf13/pflag/string.go:37 +0x88
k8s.io/cli-runtime/pkg/genericclioptions.(*ConfigFlags).AddFlags(0x140002c9200, 0x0?)
	/Users/kiki/workspace/golang/src/github.com/karmada-io/karmada/vendor/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go:327 +0x58
github.com/karmada-io/karmada/pkg/karmadactl.NewKarmadaCtlCommand({0x106049234, 0xa}, {0x106049234, 0xa})
	/Users/kiki/workspace/golang/src/github.com/karmada-io/karmada/pkg/karmadactl/karmadactl.go:55 +0x23c
main.main()
	/Users/kiki/workspace/golang/src/github.com/karmada-io/karmada/cmd/karmadactl/karmadactl.go:11 +0x34
exit status 2
```

Reply:
I just tested it that seems no problems. How do you move it?

My test is shown below
```
func NewKarmadaCtlCommand(cmdUse, parentCommand string) *cobra.Command {

....
	f := util.NewFactory(defaultConfigFlags)
....
                        Commands: []*cobra.Command{
				NewCmdLogs(f, parentCommand, ioStreams),
				NewCmdExec(karmadaConfig, parentCommand, ioStreams),
				NewCmdDescribe(karmadaConfig, parentCommand, ioStreams),
			},
```
```
func NewCmdLogs(f util.Factory, parentCommand string, streams genericclioptions.IOStreams) *cobra.Command {
```

Reply:
> I just tested it that seems no problems. How do you move it?
> 

I make the defaultConfigFlags as global command-line options in the `karmadactl.go`.